cocos/ui/UITextField.cpp

```diff
@@ -466,11 +469,17 @@ void TextField::setFontName(const std::string& name)
 {
     if(FileUtils::getInstance()->isFileExist(name))
     {
-        TTFConfig config = _textFieldRenderer->getTTFConfig();
-        config.fontFilePath = name;
-        config.fontSize = _fontSize;
-        _textFieldRenderer->setTTFConfig(config);
-        _fontType = FontType::TTF;
+        if (name.find(".fnt") != std::string::npos) {
+            _textFieldRenderer->setBMFontFilePath(name);
+            _fontType = FontType::BMFONT;
+        }
```

Comment:
consider the letters: `.fnt` case is more perfect, like this:
```C++
            std::string lowerCasedFontName = fontName;
            std::transform(lowerCasedFontName.begin(), lowerCasedFontName.end(), lowerCasedFontName.begin(), ::tolower);
            if (lowerCasedFontName.find(".fnt") != std::string::npos)
            {
                _tabLabelRender->setBMFontFilePath(fontName);
                _fontType = FontType::BMFONT;
            }
```

Reply:
Was thinking about this, but checking with npos is perhaps not the way to go. For instance, what if a font has a name like "some.fnt.ttf". Not likely to happen, but it could. Perhaps I could do:

```
std::string lcName = name;
std::transform(lcName.begin(), lcName.end(), lcName.begin(), ::tolower);
if(lcName.substr(lcName.length() - 4) == ".fnt") {
    //etc
```
What do you think?

Reply:
agree, this is more strict, `lowerCasedFontName.find(".fnt") != std::string::npos` is old logic to judge the type by font name, you can find it in other source files.